config/routes.rb

```diff
@@ -0,0 +1,9 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  root 'grub#new'
+  get '/recipe', to: 'grub#index', as: 'grub_list'
+  get '/recipes/:uri', to: 'grub#show', as: 'grub'
+  get '/recipe/new', to: 'grub#new'
```

Comment:
Two things here:

- I'm not sure that `new` is a good description of what this route is doing - something non-RESTful like `search` might be a better name
- Could you use `resources` for the index and show routes?